src/pybind/mgr/cephadm/module.py

```diff
@@ -2539,6 +2539,12 @@ def _apply(self, spec: GenericSpec) -> str:
     def apply_tuned_profiles(self, specs: List[TunedProfileSpec]) -> str:
         outs = []
         for spec in specs:
+            if spec.placement is not None:
+                all_hosts = [h.hostname for h in self.inventory.all_specs()]
+                invalid_hosts = [h.hostname for h in spec.placement.hosts if h.hostname not in all_hosts]
+                if invalid_hosts:
```

Comment:
As I understand it, this is only validating against specs that provide explicit hosts, not labels, patterns etc. I feel like we ought to be making use of the existing placement validation https://github.com/ceph/ceph/blob/main/src/pybind/mgr/cephadm/schedule.py#L176 like we do for services https://github.com/ceph/ceph/blob/main/src/pybind/mgr/cephadm/module.py#L2689-L2697. The only flaw is we don't necessarily have a valid spec to pass here (we need an actual service spec, not a tuned profile one), but I think we should be able to make one. Perhaps a custom container spec with all default values except we carry over the placement we have here?

Reply:
Fixed